includes/Settings.php

```diff
@@ -1008,6 +1012,182 @@ public function move_setting_after_id( $settings, $insert_settings, $after_setti
 		return $new_settings;
 	}
 
+	public function update_documents_settings(): void {
+		$documents = WPO_WCPDF()->documents->get_documents( 'all' );
+
+		foreach ( $documents as $document ) {
+			$document = str_replace( '-', '_', $document->get_type() );
+			add_filter( 'wpo_wcpdf_settings_fields_documents_' . $document, array( $this, 'apply_settings_categories' ), 999, 4 );
```

Comment:
This is not the correct filter, you should use this: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/a4e63addbf0abcbe25d5655287d4b8e4ad4bf005/includes/Documents/Invoice.php#L194

Maybe this filter needs to be updated for the Packing Slip.

Reply:
Maybe in the future, we can update the `init_settings()` method in the `OrderDocument` class to something like this:

```php
public function init_settings() {
	do_action( "wpo_wcpdf_before_{$this->type}_init_settings", $this );

	foreach ( $this->output_formats as $output_format ) {
		$page            = $option_group = $option_name = "wpo_wcpdf_documents_settings_{$this->get_type()}";
		$settings_fields = is_callable( array( $this, "get_{$output_format}_settings_fields" ) ) ?
			$this->{"get_{$output_format}_settings_fields"}( $option_name ) :
			array();

		// Apply legacy filter.
		$settings_fields = apply_filters( "wpo_wcpdf_settings_fields_documents_{$this->get_type()}", $settings_fields, $page, $option_group, $option_name );

		// Allow plugins to alter settings fields.
		$settings_fields = apply_filters( "wpo_wcpdf_settings_fields_documents_{$this->type}_{$output_format}", $settings_fields, $page, $option_group, $option_name, $this );

		if ( ! empty( $settings_fields ) ) {
			WPO_WCPDF()->settings->add_settings_fields( $settings_fields, $page, $option_group, $option_name );
		}
	}

	do_action( "wpo_wcpdf_after_{$this->type}_init_settings", $this );
}
```

This is similar to the function implemented for the `Invoice` class. In this way, the child class only implements a getter for the settings field, and this function also gets called in the constructor of the `OrderDocument`.